http/client/src/commonMain/kotlin/org/jetbrains/packagesearch/api/v3/http/PackageSearchApiClient.kt

```diff
@@ -211,40 +399,54 @@ public class PackageSearchApiClient(
     public suspend fun searchProjects(
         request: SearchProjectRequest,
         requestBuilder: (HttpRequestBuilder.() -> Unit)?,
-    ): List<ApiProject> =
-        defaultRequest<_, List<ApiProject>>(
+    ): List<ApiProject> {
+
+        val apiProjectsCache = apiProjectCacheCollection.await()
+
+        apiProjectsCache.find(selector = SearchCacheEntry::key.name, value = request).firstOrNull()
+            ?.also { if (!it.isExpired()) return it.value }
+
+        return defaultRequest<_, List<ApiProject>>(
             method = HttpMethod.Post,
             url = endpoints.searchPackages,
             body = request,
             requestBuilder = requestBuilder,
-        )
+        ).also {
+            apiProjectsCache.updateWhere(
+                fieldSelector = ApiProjectCacheEntry::key.name,
+                fieldValue = request,
+                upsert = true,
+                update = ApiProjectCacheEntry(request, it)
+            )
+        }
+
+    }
 
     public suspend fun refreshPackagesInfo(
         request: RefreshPackagesInfoRequest,
         requestBuilder: (HttpRequestBuilder.() -> Unit)? = null,
-    ): List<ApiPackage> =
-        defaultRequest<_, List<ApiPackage>>(
-            method = HttpMethod.Post,
-            url = endpoints.refreshPackagesInfo,
-            body = request,
-            requestBuilder = requestBuilder,
-        )
+    ): List<ApiPackage> {
+        //no caches for this endpoint
```

Comment:
Please put the caches for this as well. Even tho this endpoint does not impact the outbound traffic costs, it still has to be process. Restarting the IDE, re-opening a project, changing the layout of a project and many other actions will still re-trigger this call.